### ISSUE TITLE
Remove the need to set the MY_IP env var during OH hub setup.

### ISSUE THREAD

**Issue by dabooz** [edgexfoundry-holding/open-horizon-integration#10]
Teach the ./hub/makefile how to introspect the public IP address of the machine you're running on and use that introspected IP address during generation of the config.json file. If the introspection is not successful, make will fail. Don't forget to update the doc.

Issue #8 should be done before this one.

**Comment by chainhead** [comment]
Normally, I use `curl -s httpbin.org/ip | jq -r .origin` to automatically get the external IP address. Since this approach depends on `curl` and `jq` commands, would you think, this would be a fairly platform agnostic solution? If yes, I can take a shot.

**Comment by haveroot** [comment]
Normally, I use `curl -s httpbin.org/ip | jq -r .origin` to automatically get the external IP address. Since this approach depends on `cur`l and `jq` commands, would you think, this would be a fairly platform agnostic solution? If yes, I can take a shot.

**Comment by joewxboy** [comment]
Yes, both curl and jq are required and installed, so both would be available to retrieve the external IP address.
 
But ...  see following comment, since that technique doesn't work.

**Comment by joewxboy** [comment]
However, on my test VM, it has a local subnet IP address of 172.* and an external IP of 9.* and yet running curl -s httpbin.org/ip | jq -r .origin  shows my IP address as 129.*, which I'm guessing is my NAT'ted IP address.
 
IP address for ens3:    172.x.x.x
IP address for ens7:    9.x.x.x
curl -s httpbin.org/ip | jq -r .origin129.x.x.x
 
So that's not going to work for me.
 
The makefile in hub/oh and the instructions in the directions (once PR #12 is merged) will address this, for the most part, by viewing the active networrk connections and removing known loopback addresses and known local subnets.

**Comment by dabooz** [comment]
PR #12 is now merged, so we might be able to close this issue now.

**Comment by haveroot** [comment]
When I ran the command to set `MY_IP` on an AWS EC2 instance, there is no output. I noticed that the `awk { print $2 }` does not return a string that qualifies for the subsequent `egrep`.

```bash
ubuntu@ip-172-31-20-227:~/horizon-edgex$ ifconfig | egrep 'inet ' | awk '{ print $2 }'
172.17.0.1
172.31.20.227
127.0.0.1
```
